### DISCUSSION TITLE
Using custom avatar view in chat

### DISCUSSION THREAD

**Discussion by SeanOThomas** [GetStream/stream-chat-android#1958] [Q&A] [answered]
We're using the UI kit to display a list of messages for a chat screen (so `MessageListViewModel` and the corresponding view).  Is there a way of composing our own avatar view? When a member doesn't have an `image` link, we have a standard way of displaying an empty avatar icon. Thanks!

**Comment by zsmb13** [comment, marked as answer]
You can subclass the `AvatarBitmapFactory` class to customize how avatar images are loaded. You have to set the new instance via `ChatUI.avatarBitmapFactory`.

We have some [yet-to-be-published docs about this feature here](https://github.com/GetStream/stream-chat-android/blob/5084b1528f15530782648de559d58de6d55045d5/docusaurus/docs/Android/ui/global_customizations/chat_ui_customizations/avatar_bitmap_factory.md), and the in-code documentation for `AvatarBitmapFactory` should also tell you how to override its methods.
